lunabot_nav/src/lunabot_nav/dstar.py

```diff
@@ -304,337 +239,347 @@ def update_node(self, node: list, init: bool):
+        """
+        Update and replanning: Should trigger whenever there is a new map.
+        Sets affected nodes to update, and calculates new g values (finds new path)
+        This function runs when the map changes.
 
-    def update_replan(self, prev_map):
+        Left offset and up offset are the amount of buffer added to the left and up sides of the map
+        and are used to compare the correct parts of the map.
 
-        # This function runs when map changes
+        Finally, it calculates the new path and returns the result
+        """
 
-        print("Start updateReplan")
+        rospy.loginfo("Dstar: Updating values for new map")
 
-        self.km += (
-            (self.prev_node[0] - self.current_node[0]) ** 2
-            + (self.prev_node[1] - self.current_node[1]) ** 2
-        ) ** 0.5
         # Add to the accumulation value the distance from the last point (of changed map) to the current point
+        self.km += ((self.prev_node[0] - self.current_node[0]) ** 2
+                   + (self.prev_node[1] - self.current_node[1]) ** 2) ** 0.5
+        
+        
         self.prev_node = self.current_node.copy()  # update the prev_node
 
         for i in range(len(prev_map)):
             for j in range(len(prev_map[i])):
-                if (
-                    self.current_map[i][j] != prev_map[i][j]
-                ):  # for all differing values, update it and its surrounding nodes
+                # for all differing values, update it
+                if (self.current_map[i + up_offset][j + left_offset] != prev_map[i][j]):
+
+                    self.update_node([i + up_offset, j + left_offset])
 
-                    self.update_node([i, j], False)
+                    # The below block can update every surrounding node, seemingly is not needed.
 
                     # if (i > 0): #above
-                    #     self.update_node([i-1,j], False)
+                    #     self.update_node([i-1,j])
                     # if (i < len(self.node_values_list)-1): #below
-                    #     self.update_node([i+1,j], False)
+                    #     self.update_node([i+1,j])
                     # if (j > 0): #left
-                    #     self.update_node([i,j-1], False)
+                    #     self.update_node([i,j-1])
                     # if (j < len(self.node_values_list[0])-1): #right
-                    #     self.update_node([i,j+1], False)
+                    #     self.update_node([i,j+1])
+
+        # After all done updating, calculate the new path
+        return self.find_path()
+
+    def buffer_map_for_goal(self):
+        """
+        Add a buffer to the map so we can plan to the goal when it is outside the map.
+        Saves the amount of buffer to allow for correct translations between real-world coords and grid coords.
+        """
+
+        EXTRA_BUFFER = 3  # extra buffer to add to the map, needed because the real-world coordinates can be slightly off
```

Comment:
How was the value `3` picked? Will this need to be changed in the future? Imo it makes more sense for this to be a factor of resolution

Reply:
randomly- i suspect 1 might have worked also but i gave it the extra just in case.